modules/git/command.go

```diff
@@ -150,8 +152,14 @@ func CommonCmdServEnvs() []string {
 	return commonBaseEnvs()
 }
 
+var ErrBrokenCommand = errors.New("git command is command")
+
 // Run runs the command with the RunOpts
 func (c *Command) Run(opts *RunOpts) error {
+	if c.broken {
+		log.Error("git command is broken: %s", c.String())
```

Comment:
This error won't help in the way you expect it to as you do not add the invalid commands.
So, say you have `git rev-list --all-match` as base command and the user supplies `--skip=1000`, the error will simply be `git rev-list --all-match`, which tells absolutely nothing.

Reply:
The question is why end user need to see the full error message?

The full error message may contains sensitive information like server-side paths.

And the error message is only useful to developers. If there is no bug (and no malicious inputs), the command should never broken.

Reply:
Oh, I see your point ....

But adding invalid arguments seem dangerous ..... that's why I didn't add them.

Reply:
Yep, I also agree with that.
What about adding a second optional instance variable `brokenArguments` which is a `[]string` and `nil` by default?
Then we could just do `c.brokenArguments = append(c.brokenArguments, arg)`, and create a special `UnsafeToString() string {…}` function.

Reply:
Added more comments and use a slice to store broken args.